extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/RestClientAnnotationTest.java

```diff
@@ -0,0 +1,89 @@
+package io.quarkus.restclient;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.common.http.TestHTTPResourceManager;
+
+public class RestClientAnnotationTest {
+
+    static {
+        System.setProperty("io.quarkus.restclient.RestClientAnnotationTest$Client/mp-rest/url",
```

Comment:
It would be great to avoid using System properties for such tests. Using something like [this](https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/test/java/io/quarkus/arc/test/config/ConfigArrayConverterTest.java#L23) would be best

Reply:
@geoand thanks for the hint. That's exactly what I was looking for. I removed the System properties.

Reply:
Cool!